File: leetcode/0009-palindrome-number/solution.cpp
class Solution {
    public:
        bool isPalindrome(int x) {
            if (x < 0){
                return false;
            }
    
            int inputNum = x;
            long reversedNum = 0;
            while (x > 0){
                reversedNum = reversedNum * 10 + x % 10;
                x = x / 10;
            }
    
            return (reversedNum == inputNum);
        }
};

File: leetcode/0009-palindrome-number/solution.py
class Solution(object):
    def isPalindrome(self, x):
        if x < 0:
            return False

        inputNum = x
        newNum = 0
        while x > 0:
            digit = x % 10
            newNum = newNum * 10 + digit
            x = x // 10
        return newNum == inputNum